lib/src/context.dart

```diff
@@ -328,5 +328,5 @@ class UserContext {
 }
 
 extension _MapExtensions<K, V> on Map<K, V> {
-  void setOrRemove(K key, V? value) => value == null ? remove(key) : this[key] = value;
+  void setOrRemove(K key, V value) => value == null ? remove(key) : this[key] = value;
```

Comment:
Generic arguments have an implicit default `extends Object?` clause, so adding a `?` doesn't actually do anything here. Moreover, I can't find a way to force the call sites to have nullable `V`, which is unfortunate. i.e. you can still invoke it with a `String` rather than `String?`.

Reply:
I believe `null` works here because the map where the extension is being used is always `Map<String, dynamic>`, so `V` is always `dynamic` — if the context map were a `Map<String, String>`, for example, the `?` would have been more meaningful, as the analyzer would reject your code at the call site (not the extension implementation) if you passed a `String?` value.